src/main/java/frc/trigon/robot/subsystems/intake/IntakeConstants.java

```diff
@@ -60,11 +60,17 @@ public class IntakeConstants {
     static final double RUMBLE_DURATION_SECONDS = 0.6;
     static final double RUMBLE_POWER = 1;
     static final double NOTE_DETECTION_CONFIRMATION_DELAY_SECONDS = 0.6;
-    static final BooleanEvent BOOLEAN_EVENT = new BooleanEvent(
+    static final BooleanEvent HAS_NOTE_BOOLEAN_EVENT = new BooleanEvent(
             CommandScheduler.getInstance().getActiveButtonLoop(),
             () -> DISTANCE_SENSOR.getScaledValue() < NOTE_DISTANCE_THRESHOLD_CENTIMETERS
     ).debounce(NOTE_DETECTION_CONFIRMATION_DELAY_SECONDS);
     static final double NOTE_STOPPING_SECONDS = 1;
+    private static final double NOTE_COLLECTION_CURRENT = 10; //TODO: calibrate
+    static final double NOTE_COLLECTION_TIME_THRESHOLD_SECONDS = 0.25; //TODO: calibrate
+    static final BooleanEvent EARLY_NOTE_COLLECTION_DETECTION_BOOLEAN_EVENT = new BooleanEvent(
+            CommandScheduler.getInstance().getActiveButtonLoop(),
+            () -> MASTER_MOTOR.getSignal(TalonFXSignal.TORQUE_CURRENT) > IntakeConstants.NOTE_COLLECTION_CURRENT
```

Comment:
Math.abs the torque current. This may need to be supply current at some point, but this is fine for now.